### PR TITLE
Use NumTypeEnclosure for agileTree nodes

### PR DESCRIPTION
**Pull request by nkunal**
# Description
Summarize the change.

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/writer/startree_test.go [hunk 1]
```diff
@@ -388,11 +388,11 @@ func TestStarTreeMedium(t *testing.T) {
 
 		// first TotalMeasFns will be for col "e"
 		agSumIdx := 1*(TotalMeasFns) + MeasFnSumIdx
-
-		assert.Equal(t, root.aggValues[agSumIdx].CVal.(int64),
+		fmt.Println("root.aggValues[agSumIdx]=", root.aggValues[agSumIdx])
+		assert.Equal(t, root.aggValues[agSumIdx].IntgrVal,
```
Comment (AndrewHess) [inline pkg/segment/writer/startree_test.go:392]: nit: do we want this print in this test, or did you forget to remove it?

File: pkg/segment/writer/agiletree.go [hunk 2]
```diff
@@ -188,13 +188,28 @@ func (stb *StarTreeBuilder) resetNodeData() {
...
+	// and after that if the nodePool count exceeds then we can do the
+	// one by one extension
+	sizeNeeded := MaxAgileTreeNodeCount - len(stb.nodePool)
+	if sizeNeeded > 0 {
+		newArr := make([]*Node, sizeNeeded)
+		for i := 0; i < sizeNeeded; i++ {
+			newArr[i] = &Node{}
+		}
+		stb.nodePool = append(stb.nodePool, newArr...)
+	}
```
Comment (AndrewHess) [inline pkg/segment/writer/agiletree.go:212]: nit: there's a util for this
```suggestion
stb.nodePool = toputils.ResizeSlice(stb.nodePool, MaxAgileTreeNodeCount)
```
For the case `sizeNeeded < 0` in your original code, I'm pretty sure the extra space in the slice will not get garbage collected (`ResizeSlice()` doesn't decrease the capacity, and I'm pretty sure the GC won't touch anything that's in the capacity but not the length, but I haven't been able to find documentation for this)

Same comment elsewhere

**Review by AndrewHess — COMMENTED** [review]
Looks good; minor comments

File: pkg/segment/writer/startree_test.go (hunk 1 above)
Comment (Macbeth98) [reply pkg/segment/writer/startree_test.go:392]: I forgot to remove it.

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**